#134209517 send game invite

tomily1 opened the pull request:
#### What does this PR do?
This PR searches for users from the database and send an invites to registered and unregistered users.
#### Description of Task to be completed?
Search for users from the database and send invites with the URL for them to join the game. The invites can be sent to both registered and unregistered users.
#### How should this be manually tested?
The result of the test can be seen by running the test and viewing the result of the api routes created.
#### Any background context you want to provide?
No background context
#### What are the relevant pivotal tracker stories?
Users should be able to send invites to at most 11 people other than themselves
#### Screenshots (if appropriate)
![screen shot 2016-12-05 at 6 34 24 pm](https://cloud.githubusercontent.com/assets/19914657/20895382/850704bc-bb19-11e6-9375-bced312ef23a.png (178KB))
#### Questions:
No questions.

Commit:
bug(fix-travis-issues): set travis env variable for mongohq_url to make test pass.

Commit:
bug(add-files): add public/lib files to the aplication

remove the public/lib folder from .gitignore file to allow images and icon to be displayed on the application when launched

#finish

Commit:
bug(add-files): add public/lib files to the aplication

add .eslintignore to the hound.yml file to ignore libraries from being flagged

#finish

Commit:
bug(add-files): add public/lib to the eslintignore file

add .eslintignore to the hound.yml file to ignore libraries from being flagged

#finish

Commit:
bug(add-files): add public/lib to the eslintignore file

add .eslintignore to the hound.yml file to ignore libraries from being flagged

#finish

Commit:
bug(fix-public-lib-folder): fix issue with public/lib not being created on heroku

fix issue with public/lib not being created on deploying to heroku. change the gulp bower task and use gulp-bower-files

finish

Commit:
bug(fix-loading-issue): fix landing page loading issues

Fix issues with landing page and other pages not loading not loading on heroku.
Add bower to package.json and create a task on gulp that transfers the file into
public/lib

finish

Commit:
bug(fix-style-issues): rewrite gulpfile to es6 format

finishes

Commit:
bug(heroku-deployment): remove the public/lib folder from being tracked

finish

Commit:
bug(change-to-eslint): change jshint to eslint in the gulp task

finish

Commit:
bug(change-to-eslint): edit gulp file and jade template files

finish
edit the link to script and stylesheets in the jade templates

finish

Commit:
bug(edit-gulpfile): fix issue with unresolved path in gulpfile

finish

Commit:
bug(edit-files): remove white spaces and console.log statements

finish

Commit:
bug(fix-houndci-issues): fix issues with houndCI and modules path not being resolved

finish

Commit:
bug(fix-houndci-issues): fix issues with houndCI and modules path not being resolved

finish

Commit:
bug(fix-houndci-issues): fix issues with houndCI and modules path not being resolved

finish

Commit:
bug(fix-houndci-issues): fix issues with houndCI and modules path not being resolved

finish

Commit:
bug(gulp-sass): add gulp-sass to the dependencies in package.json

finish

Commit:
bug(gulp-sass): add gulp-sass to the dependencies in package.json

finish

Commit:
bug(heroku-deployment): add gulp build to package.json to run gulp task on heroku

finish

Commit:
bug(heroku-deployment): convert angular link from http to https

convert http to https because the files could not be served in http

finish

Commit:
feature(Search-API-Test): Write Test for Search API

Commit:
feature(Search-API-Test): Write Test for Search API

Commit:
test(search-api-test): write test for the search api

Commit:
test(search-api-test): write test for search api

Commit:
feature(Search-API-Test): Write Test for Search API

Commit:
updated socket.io

Commit:
updated socket.io

Commit:
updated socket.io

Commit:
fixes(review-app): create an app.json file to be used for review app on heroku

finish

Commit:
written test for the search api

Commit:
test(search-api): write tests for the search api

Commit:
written test for the search api

Commit:
feature(search-for-users): writing tests for the search api

Commit:
feature(search-for-users): writing tests for the search api

Commit:
feature(search-for-users): writing tests for the search api

Commit:
feature(search-for-users): writing tests for the search api

Commit:
bug:change travis settings to make build pass

Commit:
fix(merge-conflict): fix merge conflict from staging branch

finish #134209517
Merge branch 'staging' of github.com:andela/kissa-cfh-project into feature/134209517/invite

Commit:
feature(nodemailer):send invites using nodemailer

write a function that sends emails for users to join game

finishes #134209517

Commit:
feature(nodemailer):send invites using nodemailer

write a function that sends emails for users to join game

finishes #134209517

Commit:
feature(nodemailer):send invites using nodemailer

write a function that sends emails for users to join game

finishes #134209517

Commit:
feature(search-users): create route that searches for users

create function that searches the users from the database

finishes #134209517

Commit:
feature(search-users): create route that searches for users

create function that searches the users from the database

finishes #134209517

Commit:
feature(search-users): create route that searches for users

create function that searches the users from the database

finishes #134209517

Commit:
feature: reroute user to the previous page

reroute users to the previous page on successful login or signup

finishes #134209517

Commit:
feature: reroute user to the previous page

reroute users to the previous page on successful login or signup

finishes #134209517

Commit:
refactor:remove localhost url for mongodb

fixes #134209517

Commit:
refactor:remove localhost url for mongodb

fixes #134209517

Commit:
feature(maximum-player): increase maximum number of players

increase maximum numbers of players to 12 and also add player's id to player object

finishes #134209517

Commit:
feature(maximum-player): increase maximum number of players

increase maximum numbers of players to 12 and also add player's id to player object

finishes #134209517

Commit:
feature: add player id for game with friends

add player id if user is joining a game with friends

finishes #134209517

Commit:
feature: add player id for game with friends

add player id if user is joining a game with friends

finishes #134209517

Commit:
feature: create search and invite route

add the search and invite route to the routes.js file

finishes #134209517

Commit:
feature: create search and invite route

add the search and invite route to the routes.js file

finishes #134209517

Commit:
feature: create search and invite route

add the search and invite route to the routes.js file

finishes #134209517

Commit:
test(invite-user): create invite user test suite

write test for user invite route

finishes #134209517

Commit:
test(invite-user): create invite user test suite

write test for user invite route

finishes #134209517

Commit:
test(invite-user): create invite user test suite

write test for user invite route

finishes #134209517

Commit:
test(search-user): create search user test suite

write test for user search route

finishes #134209517

Commit:
test(search-user): create search user test suite

write test for user search route

finishes #134209517

Commit:
test(search-user): create search user test suite

write test for user search route

finishes #134209517

Commit:
fix(coveralls): increase coveralls percentage

increase test coverage by 10% by editing game.js test file

fixes #134209517

Commit:
fix(coveralls): increase coveralls percentage

increase test coverage by 10% by editing game.js test file

fixes #134209517

Commit:
fix(coveralls): increase coveralls percentage

increase test coverage by 10% by editing game.js test file

fixes #134209517

Commit:
fix(mocha-error): add use strict to search.js test file

remove flagged mocha error by adding use strict for let keyword

fixes #134209517

Commit:
fix(mocha-error): add use strict to search.js test file

remove flagged mocha error by adding use strict for let keyword

fixes #134209517

Commit:
fix(mocha-error): add use strict to search.js test file

remove flagged mocha error by adding use strict for let keyword

fixes #134209517

Commit:
feature(invites): show modal popup and search input to invite users

show text input to search and invite users and also show modal pop up when number of players is not up to minimum required number or exceeds maximum number of invites

finishes #134209517

Commit:
feature(invites): show modal popup and search input to invite users

show text input to search and invite users and also show modal pop up when number of players is not up to minimum required number or exceeds maximum number of invites

finishes #134209517

Commit:
feature(sign-in): update email value in login form

updates email value when unregistered user tries to access a game and is redirected to sign up page

finishes #134209517

Commit:
feature(sign-in): update email value in login form

updates email value when unregistered user tries to access a game and is redirected to sign up page

finishes #134209517

Commit:
fix(eslint): add use strict for let keyword

remove flagged eslint error by adding use strict for let keyword

fixes #134209517

Commit:
fix(eslint): edit search file

remove flagged eslint error by adding use strict for let keyword

fixes #134209517

Commit:
fix(eslint): edit search file

remove flagged eslint error by adding use strict for let keyword

fixes #134209517

Commit:
feature(game): change the maximum number of players

change the maximum numbers of players in a game from 6 to 12

finishes #134209517

Commit:
feature(game): change the maximum number of players

change the maximum numbers of players in a game from 6 to 12

finishes #134209517

Commit:
feature(invite): add directive for invite list template

Add directive to display list of invited players

finishes #134209517

Commit:
feature(invite): add directive for invite list template

Add directive to display list of invited players

finishes #134209517

Commit:
feature(search): add search box and invite button to game screen UI

Add the Interface to search and invite user on the HTML file

finishes #134209517

Commit:
feature(search): add search box and invite button to game screen UI

Add the Interface to search and invite user on the HTML file

finishes #134209517

Commit:
feature(search): add search box and invite button to game screen UI

Add the Interface to search and invite user on the HTML file

finishes #134209517

Commit:
feature(invite): create invite template

create template that displays lists of invited users to a game with friends

finishes #134209517

Commit:
feature(invite): create invite template

create template that displays lists of invited users to a game with friends

finishes #134209517

Commit:
feature(popup): create popup template

create popup template that alerts users when the minimum amount of players has not been
reached or the maximum amount of players have been exceeded.

finishes #134209517

Commit:
feature(popup): create popup template

create popup template that alerts users when the minimum amount of players has not been
reached or the maximum amount of players have been exceeded.

finishes #134209517

Commit:
feature(game): show start button only to game creator

hide button to users who are not game creator and show to the game creator only

finishes #134209517

Commit:
feature(game): show start button only to game creator

hide button to users who are not game creator and show to the game creator only

finishes #134209517

Commit:
feature(authentication): add user's email to authentication form on redirect

add user's email to ease authentication when an unauthenticated user tries to join a game with friends

finishes #134209517

Commit:
feature(authentication): add user's email to authentication form on redirect

add user's email to ease authentication when an unauthenticated user tries to join a game with friends

finishes #134209517

Commit:
chore(modules): add nodemailer module and delete faker

add useful dependencies and remove dependencies that are not required

finishes #134209517

Commit:
chore(modules): add nodemailer module and delete faker

add useful dependencies and remove dependencies that are not required

finishes #134209517

Commit:
chore(modules): add nodemailer module and delete faker

add useful dependencies and remove dependencies that are not required

finishes #134209517

Commit:
chore(modules): add public/js folder to eslintignore

add files in public/js folder to the eslintignore file

finishes #134209517

Commit:
chore(modules): add public/js folder to eslintignore

add files in public/js folder to the eslintignore file

finishes #134209517

Commit:
chore(modules): add public/js folder to eslintignore

add files in public/js folder to the eslintignore file

finishes #134209517

Commit:
fix(bower): upgrade dependencies on bower

upgrade angular bootstrap from 0.4.0 to 0.8.0 on bower

finishes #134209517

Commit:
fix(bower): upgrade dependencies on bower

upgrade angular bootstrap from 0.4.0 to 0.8.0 on bower

finishes #134209517

Commit:
feature(styling): write style code for search UI

write the style code for the search UI template

finishes #134209517

Commit:
feature(styling): write style code for search UI

write the style code for the search UI template

finishes #134209517

Commit:
feature(styling): write style code for search UI

write the style code for the search UI template

finishes #134209517

Commit:
fix(eslint): set eslint configuration

set the eslint config for api test files

fixes #134209517

Commit:
fix(eslint): set eslint configuration

set the eslint config for api test files

fixes #134209517

Commit:
fix(eslint): set eslint configuration

set the eslint config for api test files

fixes #134209517

andela-jare commented on app/controllers/api/invites.js:
```diff
@@ -0,0 +1,31 @@
+const nodemailer = require('nodemailer');
+require('dotenv').config({ silent: true });
+
+exports.emailinvite = (req, res) => {
```
Don't use abbreviation

Commit:
feature(game-ui): move search input and invite button to game display link box

move the search input and invite button to where the game invite link is located for better user experience

finishes #134209517

Commit:
fix(merge-conflict): fix merge conflict on eslintignore file

Commit:
fix(merge-conflict): fix merge conflict on eslintignore file

Commit:
"fix(merge-conflict): fix merge conflict on eslintignore file"

Commit:
Revert "fix(merge-conflict): fix merge conflict on eslintignore file"

This reverts commit 444d8c6fcde18af525984bd799c5a7377fa459ff, reversing
changes made to 2a14666721fac40737fa11c32996a83c246c2381.

Commit:
Revert "Revert "fix(merge-conflict): fix merge conflict on eslintignore file""

This reverts commit ec78cd7cad02e9da9bd92a1cfeb6bc3d5eef3061.

Commit:
Revert "fix(merge-conflict): fix merge conflict on eslintignore file"

This reverts commit 444d8c6fcde18af525984bd799c5a7377fa459ff, reversing
changes made to 2a14666721fac40737fa11c32996a83c246c2381.

Commit:
Revert "Revert "fix(merge-conflict): fix merge conflict on eslintignore file""

This reverts commit ec78cd7cad02e9da9bd92a1cfeb6bc3d5eef3061.

Commit:
fix(rebase): fix commit messages and merge conflicts during rebase

Commit:
fix(merge-conflict): fix merge conflict when pulling remote branch
Merge branch 'feature/134209517/send-game-invite' of https://github.com/andela/kissa-cfh-project into feature/134209517/send-game-invite

Commit:
fix(houndCI): fix the error flagged by houndCI as a result of merge conflict
finish

Commit:
fix(travis): fix travis issue as a result of merge conflict
finish

ngozi-ekekwe commented on app/controllers/api/search.js:
```diff
@@ -0,0 +1,27 @@
+          return res.json(err);
+        } else if (!result) {
+          // hope this never happens.
+          return res.send('I did not find any data');
```
return a status code along with the response

vonvick replied:
will work on it.

tomily1 commented:
This PR is closed because package.json is conflicting with the staging as result of mongoose.
